### PR TITLE
Design Profile / Patient entities relations

### PR DESCRIPTION
**Pull request by vladimir-paziuk**

### CODE REVIEW

File: src/app.module.ts [hunk 1]
```diff
@@ -2,14 +2,23 @@ import { Module } from '@nestjs/common';
...
 
-import { AuthModule } from './auth/auth.module';
-import { DoctorsModule } from './doctors/doctors.module';
-import { UserEntity } from './auth/user.entity';
-import { DoctorEntity } from './doctors/doctor.entity';
+import { AuthModule } from 'auth/auth.module';
+import { UserEntity } from 'auth/user.entity';
+
+import { DoctorsModule } from 'doctors/doctors.module';
+import { DoctorEntity } from 'doctors/doctor.entity';
```
Comment (macc4) [inline src/app.module.ts:9]: why absolute imports?

File: src/common/constants/base.entity.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  @CreateDateColumn({
+    name: 'created_at',
+    type: 'timestamp',
+    default: () => 'CURRENT_TIMESTAMP(6)',
+  })
+  createdAt: string;
+  @UpdateDateColumn({
+    name: 'updated_at',
+    type: 'timestamp',
+    nullable: true,
```
Comment (macc4) [inline src/common/constants/base.entity.ts:14]: does it need nullable? it's gonna always equal to current timestamp

File: src/patients/dtos/patient.dto.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsEnum, IsUUID } from 'class-validator';
+import { BLOOD_TYPE_ENUM } from 'common/constants/blood-type.enum';
+
+export class PatientDto {
+  @ApiProperty()
```
Comment (macc4) [inline src/patients/dtos/patient.dto.ts:6]: @ApiProperty({
    enum: BLOOD_TYPE_ENUM,
  })

File: src/patients/patients.service.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      return entity;
+    }
+    throw new NotFoundException();
+  }
+
+  async updatePatient(id: string, dto: PatientDto): Promise<PatientEntity> {
+    const entity = await this.getPatientById(id);
+    const updated = { ...entity, ...dto };
+
+    await this.patientsRepository.save(updated);
```
Comment (macc4) [inline src/patients/patients.service.ts:49]: you are performing two separate requests here, isn't there a more efficient way with an update query?

File: src/profiles/profiles.service.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    throw new NotFoundException();
+  }
+
+  async createProfile(profile: ProfileCreateDto): Promise<ProfileEntity> {
+    const entity = await this.profilesRepository.createProfile(profile);
+
+    if (entity) {
+      return entity;
+    }
+    throw new NotFoundException();
```
Comment (macc4) [inline src/profiles/profiles.service.ts:42]: not found exception on create?

File: src/profiles/profiles.service.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  async updateProfile(id: string, dto: ProfileDto): Promise<ProfileEntity> {
+    const entity = await this.getProfileById(id);
+    const updated = { ...entity, ...dto };
+
+    await this.profilesRepository.save(updated);
+
+    return updated;
```
Comment (macc4) [inline src/profiles/profiles.service.ts:51]: should be a not found check here

File: src/profiles/profiles.service.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  }
```
Comment (macc4) [inline src/profiles/profiles.service.ts:52]: also two separate requests
you should try updating an existing row with a where query, and if nothing is updated - then you can throw a not found exception, without searching for the entity first

File: src/app.module.ts (hunk 1 above)
Comment (vladimir-paziuk) [reply src/app.module.ts:9]: For using dependencies inside of module I use relative path for keep modularity.
For access to outside world better to use absolute paths, its mush simple for refactoring or transferring

@macc4 Agree, for app.module.ts it doesn't have a lot of sense

File: src/main.ts [hunk 8]
```diff
@@ -1,12 +1,14 @@
...
 import { ValidationPipe } from '@nestjs/common';
-import { initSwagger } from './common/swagger/swagger.config';
+import { initSwagger } from 'common/swagger/swagger.config';
+import { TransformInterceptor } from 'common/transform.interceptor';
 
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
 
   app.useGlobalPipes(new ValidationPipe());
+  app.useGlobalInterceptors(new TransformInterceptor());
```
[the remark this answers is not in the document]
Comment (vladimir-paziuk) [reply src/main.ts:11]: @macc4 I am not sure, it's instance of class-transformer library which I use for exclude value from response 
image

**Pull request merged**